python/ray/data/_internal/execution/operators/base_physical_operator.py

```diff
@@ -79,6 +80,11 @@ def num_outputs_total(self) -> Optional[int]:
             else self.input_dependencies[0].num_outputs_total()
         )
 
+    def num_output_rows_total(self) -> Optional[int]:
+        if self._num_outputs:
+            return sum(bundle.num_rows() for bundle in self._output_buffer)
+        return self.input_dependencies[0].num_output_rows_total()
```

Comment:
i think here, we can simply return the input dependency's `num_output_rows_total()`. Because we don't know when/if `self._output_buffer` is completed, this may give a lower total row count as the AllToAllOperator continues to execute and more items are added into `self._output_buffer`.

```suggestion
        return self.input_dependencies[0].num_output_rows_total()
```

Reply:
fixed!